### ISSUE TITLE
zabapgit program replaced

### ISSUE THREAD

**Issue by A141117** [larshp/abapGit#1231]
When I installed the abapgit_repo from the ZABAPGIT transaction then it has overwritten the program ZABAPGIT which I initially copied from the abapcode notepad file on github website.  So which one should I keep - the notepad one or the smaller one which came later after abapgit_repo installation?

**Comment by larshp** [comment]
the smaller one, it is the development version which can easily be upgraded via pull

**Comment by A141117** [comment]
But your wiki says that copy abapcode form that notepad file. So whats that code is for?

**Comment by larshp** [comment]
see http://docs.abapgit.org/guide-upgrade.html

it makes it easier to upgrade

plus easier to contribute changes, as the smaller matches the code stored in this git repository

**Comment by A141117** [comment]
So for further upgrades I need not to copy abap program from github and it will be done automatically?

**Comment by A141117** [comment]
As per my organization requirement i want to deactivate below functionalities:
Transport to branch option , Update local checksum option and Uninstall. 
Earlier I commented  big program and now I'm unable to find the related code in class.

**Comment by A141117** [comment]
'Transport to branch' option will overwrite the existing code and 'Uninstall function' will simply delete the abap objects from cloned repositories. So we wanted to disable these functionalities until we have a proper security checks/roles for both of these functionalities. 
I'm fine if 'Update local checksum' will just updates the state for the repo. 
So can you please tell me the class method where you are building the menu to show these options "Transport to branch" and "Uninstall" in abapgit main screen. I want to disable these functionalities by making them invisible from menu.

**Comment by A141117** [comment]
Regarding the write protected option you suggested above, in that case I have to get access to each repository for our organization and make this write protected changes for each one which seems not feasible. I cannot keep the track when are the new repositories getting created.

**Comment by larshp** [comment]
All 3 are defined in:
https://github.com/larshp/abapGit/blob/master/src/ui/zcl_abapgit_gui_view_repo.clas.abap

we should some time add these 3 functions to the authorization concept(see https://github.com/larshp/abapGit/blob/master/src/zcl_abapgit_auth.clas.abap) so the abapGit code in your system can say identical to this repo, and you'll have a few add-ons which defines the extra authorization behavior

**Comment by larshp** [comment]
TLDR, implement authorization checks for: 
* Transport to branch option
* Update local checksum option
* Uninstall

**Comment by larshp** [comment]
@A141117 will this work? see https://github.com/larshp/abapGit/pull/1269
